tests/integration.rs

```diff
@@ -5396,6 +5410,125 @@ mod tests {
+                    .args(&["create", "-f", "raw", &img.to_str().unwrap(), "100m"])
+                    .output()
+                    .expect("qemu-img command failed")
+                    .stdout;
+                println!("{:?}", out);
+
+                // Associate image to a loop device
+                let out = Command::new("losetup")
+                    .args(&["-f", &img.to_str().unwrap()])
+                    .output()
+                    .expect("failed to create loop device")
+                    .stdout;
+                println!("{:?}", out);
+
+                // Determine the associated loop device name
+                // LOOP=$(sudo losetup -a | grep -e "${IMG}" | cut -d: -f1)
+                let out = Command::new("losetup")
+                    .args(&["-a"])
+                    .output()
+                    .expect("loop device not found")
+                    .stdout;
+                let _tmp = String::from_utf8_lossy(&out as &[u8]);
+                let m: Vec<&str> = _tmp.matches(&img.to_str().unwrap()).collect();
+                if m.len() < 1 {
+                    return String::new();
+                }
+                let _tmp = String::from_utf8_lossy(m[0].as_bytes());
+                let loop_dev = _tmp.split(":").collect::<Vec<&str>>()[0];
+
+                // Create a partition table
+                // echo 'type=7' | sudo sfdisk "${LOOP}"
+                let mut child = Command::new("sfdisk")
+                    .args(&[loop_dev])
+                    .stdin(Stdio::piped())
+                    .spawn()
+                    .unwrap();
+                let stdin = child.stdin.as_mut().expect("failed to open stdin");
+                let _ = stdin
+                    .write_all("type=7".as_bytes())
+                    .expect("failed to write stdin");
+                let out = child.wait_with_output().expect("sfdisk failed").stdout;
+                println!("{:?}", out);
+
+                // Disengage the loop device
+                let out = Command::new("losetup")
+                    .args(&["-d", &loop_dev])
+                    .output()
+                    .expect("loop device not found")
+                    .stdout;
+                println!("{:?}", out);
+
+                // Re-associate loop device pointing to the partition only
+                let out = Command::new("losetup")
+                    .args(&[
+                        "--offset",
+                        (512 * 2048).to_string().as_str(),
+                        "-f",
+                        &img.to_str().unwrap(),
+                    ])
+                    .output()
+                    .expect("failed to create loop device")
+                    .stdout;
+                println!("{:?}", out);
+
+                // Create msdos filesystem.
+                // XXX mkfs.ntfs is missing in the docker image and should be added
+                let out = Command::new("mkfs.msdos")
```

Comment:
Why are you bothering creating a filesystem since you're not checking the content of it from the integration test itself?

Reply:
Yeah, you're right. It as on my plan, just seemed to much for this one as i previously tested manually. I've added a piece to write/read a file from the mounted volume now for it to make sense.

thanks.

Reply:
Cool!